typing/typecore.ml

```diff
@@ -6911,7 +6879,13 @@ and type_let
                                       | _ -> false) pat_extra) then
             check_partial_application ~statement:false vb_expr
       | _ -> ()) l;
-  (l, new_env, unpacks)
+  (* Note that we take care to avoid calling [add_module_variables]
```

Comment:
Ah. You've already written the comment I wanted. But it needs to be in two places. I am a huge fan of GHC's `Note` convention, where this might become

```
Note [add_module_variables after checking expressions]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<< text here >>
```

and then both here and in `type_pattern_list`, there is `See Note [add_module_variables after checking expressions]`.

Reply:
See above: I adopted something like this convention, but the note is localized to `type_cases`. (Did you have another place in mind for the parent note?)